From a170caa8c68b45691fd3fd7ca0f054c3382d7577 Mon Sep 17 00:00:00 2001
From: JackNoordhuis <daswaggamer@gmail.com>
Date: Fri, 2 Mar 2018 21:32:44 +1100
Subject: [PATCH] Name change


diff --git a/src/pocketmine/PocketMine.php b/src/pocketmine/PocketMine.php
index 5e82a11..a1085cf 100644
--- a/src/pocketmine/PocketMine.php
+++ b/src/pocketmine/PocketMine.php
@@ -79,7 +79,7 @@ namespace pocketmine {
 	use pocketmine\wizard\SetupWizard;
 	use raklib\RakLib;
 
-	const NAME = "PocketMine-MP";
+	const NAME = "PocketMine-MP-Hybrid";
 	const VERSION = "1.7dev";
 	const API_VERSION = "3.0.0-ALPHA11";
 	const CODENAME = "[REDACTED]";
-- 
2.6.4

